### PR TITLE
WIP SPIDER: CPSBOE

### PR DESCRIPTION
**Pull request by mwgalloway**

### CODE REVIEW

File: documenters_aggregator/spiders/cpsboe.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                'description': self._parse_description(item),
+                'classification': self._parse_classification(item),
+                'start_time': self._parse_start_time(item),
+                'all_day': self._parse_all_day(item),
+                'status': self._parse_status(item),
+                'location': self._parse_location(item)
+
+            }
+
+    def _parse_id(self, item):
```
Comment (eads) [inline documenters_aggregator/spiders/cpsboe.py]: Clever! Are you pretty sure there are never two events on the same day? (I'm guessing this is a safe assumption in this case, but worth asking.)

File: documenters_aggregator/spiders/cpsboe.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        text_list = self._remove_line_breaks(item.css('::text').extract())
+        split_date_string = text_list[0].split()[:3]
+        split_date_string[0] = split_date_string[0][:3]
+        split_date_string[1] = split_date_string[1][:-1]
+        date_string = " ".join(split_date_string)
+        date = datetime.strptime(date_string, '%b %d %Y')
+        return str(10000*date.year + 100*date.month + date.day)
+
+    def _remove_line_breaks(self, collection):
+        while '\n' in collection: collection.remove("\n")
```
Comment (eads) [inline documenters_aggregator/spiders/cpsboe.py]: This is very elegant.

File: tests/test_cpsboe.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import pytest
+
+from tests.utils import file_response
+from documenters_aggregator.spiders.cpsboe import CpsboeSpider
+
+test_response = file_response('files/cpsboe.html')
+spider = CpsboeSpider()
+parsed_items = [item for item in spider.parse(test_response) if isinstance(item, dict)]
+
+assert len(parsed_items) == 14
```
Comment (eads) [inline tests/test_cpsboe.py]: This should be inside a function named something like `test_items_length()`.

**Comment by eads** [comment]
Code looks very good except for the floating assertion in the test file.

However, there are some style things to be taken care of:

```
$ pep8 --ignore E265,E266,E501 .
./documenters_aggregator/spiders/cpsboe.py:7:1: E302 expected 2 blank lines, found 1
./documenters_aggregator/spiders/cpsboe.py:39:25: E226 missing whitespace around arithmetic operator
./documenters_aggregator/spiders/cpsboe.py:39:41: E226 missing whitespace around arithmetic operator
./documenters_aggregator/spiders/cpsboe.py:42:33: E701 multiple statements on one line (colon)
./documenters_aggregator/spiders/cpsboe.py:46:4: W291 trailing whitespace
./documenters_aggregator/spiders/cpsboe.py:82:5: E303 too many blank lines (2)
./documenters_aggregator/spiders/cpsboe.py:93:1: W391 blank line at end of file
./tests/test_cpsboe.py:13:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:17:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:20:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:24:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:31:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:35:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:38:1: E302 expected 2 blank lines, found 1
./tests/test_cpsboe.py:53:6: W292 no newline at end of file
```

I'll add some affordances to make this testing more visible/accessible before you make your PR, but it's good to keep the code clean.

File: documenters_aggregator/spiders/cpsboe.py (hunk 1 above)
Comment (shua123) [inline documenters_aggregator/spiders/cpsboe.py]: There should only be one event in a day on the planning calendar for regular board meetings. If you get to actual/all board meetings, there can be multiple meetings on the same day ([committee meetings](http://www.cpsboe.org/meetings/details/238), [public](http://www.cpsboe.org/meetings/details/212) hearings, [budget](http://www.cpsboe.org/meetings/details/251) meetings, etc).

See past meetings for examples: http://www.cpsboe.org/meetings/past-meetings

**Comment by jim** [comment]
@mwgalloway This PR looks good other than @eads's notes above re: moving an assertion in to a function and resolving style issues. You can use `invoke runtests` to run the unit tests and check your code style locally.

**Comment by eads** [comment]
oooh `./documenters_aggregator/spiders/cpsboe.py:48:13: E123 closing bracket does not match indentation of opening bracket's line`

... I'm inclined just to suppress that one.

**Pull request merged**